src/domain_extended/mod.rs

```diff
@@ -0,0 +1,198 @@
+            )?;
+            commitment.push(commitment_item);
+            if is_hiding {
+                randomness.push(randomness_item)
+            }
+        }
+
+        Ok((commitment, randomness))
+    }
+
+    /// Evaluation proof for an arbitrary degree polynomial.
+    /// Note: this default implementation demands that the state of the Fiat-Shamir
+    /// rng is already bound to the evaluation claim (i.e. the commitment of the polynomial,
+    /// the query point, and the value).
+    fn open(
+        ck: &Self::CommitterKey,
+        polynomial: Polynomial<G::ScalarField>,
+        point: G::ScalarField,
+        is_hiding: bool,
+        randomness: Self::Randomness,
+        fs_rng: &mut Self::RandomOracle,
+        rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::Proof, Self::Error> {
+        let key_len = ck.get_key_len();
+
+        if key_len.next_power_of_two() != key_len {
+            Err(Error::Other(
+                "Commiter key length is not power of 2".to_owned(),
+            ))?
+        }
+
+        let lc_time = start_timer!(|| "LC of polynomial and randomness");
+
+        let p_len = polynomial.coeffs.len();
+        let segments_count = std::cmp::max(
+            1,
+            p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 },
+        );
+
+        // Compute the query-point dependent linear combination of the segments,
+        // both for witnesses, commitments (and their randomnesses, if hiding)
+
+        let point_pow = point.pow(&[key_len as u64]);
+
+        let mut raw_poly_lc = Vec::new();
+        let mut raw_rand_lc = Vec::new();
+        for i in (0..segments_count).into_iter() {
+            raw_poly_lc.push(
+                Polynomial::from_coefficients_slice(&polynomial.coeffs[i * key_len..core::cmp::min((i + 1) * key_len, p_len)])
+            );
+
+            if is_hiding {
+                // TODO: check the situation when poly has one segment more comparing to
+                //  randomness segments
+                raw_rand_lc.push(
+                    if randomness.len() <= i {
+                        PC::Randomness::zero()
+                    } else {
+                        randomness[i].clone()
+                    },
+                );
+            }
+        }
+
+        end_timer!(lc_time);
+
+        PC::open_lc(
+            ck,
+            LinearCombination::new_from_val(&point_pow, raw_poly_lc.iter().collect()),
+            point,
+            is_hiding,
+            LinearCombination::new_from_val(&point_pow, raw_rand_lc.iter().collect()),
+            fs_rng,
+            rng,
+        )
+    }
+
+    fn succinct_verify<'a>(
+        vk: &Self::VerifierKey,
+        combined_commitment: &'a Self::Commitment,
+        point: G::ScalarField,
+        value: G::ScalarField,
+        proof: &Self::Proof,
+        fs_rng: &mut Self::RandomOracle,
+    ) -> Result<Option<Self::VerifierState>, Self::Error> {
+        let key_len = proof.get_key_len();
```

Comment:
As we have killed this function from the PCProof trait, we are going to kill this call and replace it with `vk.get_key_len()`, this is what someone would expects by reading the function signatures.
We were reading the key length out of the proof mainly for our use case (in which we could verify a batch of proofs being produced by keys with smaller sizes with respect to the one used for batch verification), but this is a very specific use case and it should be caller responsibility to trim the vk to the same size of the proof

Reply:
I will leave this comment open: we need to remember to update the batch verifier in Darlin to perform this trimming

Reply:
Not relevant anymore.